### ISSUE TITLE
Event Log

### ISSUE THREAD

**Issue by Apexal** [Daddy-s-Dungeons-Tools/ddtools-web#16]

**Comment by Apexal** [comment]
Good progress so far. 
![image](https://user-images.githubusercontent.com/8422699/170887618-7419f568-1546-4607-9824-a94e7fe02394.png)

**Comment by Apexal** [comment]
Other platforms appear to merge this in with chat so that in one place users can see "Player X rolled a d20 for a result of 17" and also "Player X: blah blah blah" and "Player X shared Item 'Adventurer's Pack'" in one place.

Some thoughts:
1. We can ignore that and go ahead with the event log and not have a chat function at all, as we have Discord and will be in voice chat.
2. We can have this event log and a separate chat but then where do things like roll results go? One or the other? Both? Not super clear.
3. We can combine events and chat and just have default filters that don't show things like "campaign-updated" to players.

**Comment by Apexal** [comment]
Nick and I discussed that chat is probably a bad idea for our group because they just meme and get distracted (sorry Aaron).

However, being able to share info with other characters via the log is useful, such as item descriptions, spells, notes, etc.

**Comment by zacharylove** [comment]
One thing to note- dice rolls should be included in the log. Currently, the result of a dice roll seems to only be visible to whoever rolled it, as shown here:
![image](https://user-images.githubusercontent.com/12072877/173936680-5e2b34f9-dee3-4cd9-b266-d8c9a0509377.png)

The results of each dice roll should be added to the log and visible to the DM, the roller, and (perhaps) the rest of the campaign's members.

Let me know if I'm missing anything, otherwise we can make this into a new issue

**Comment by Apexal** [comment]
@zacharylove Totally right, the roll button was quickly setup for testing and hasn't been connected to anything yet. We should decide though on which approach for logging dice rolls since it takes time for the dice to roll and settle on a result:
- wait for end of roll and just log the result
- immediately log "Atreyu is rolling a d20..." and then update the log message with the result once done (I think this is cooler)

**Comment by zacharylove** [comment]
Agreed, the second option would be better- easier for debugging too!
We can set up two separate log entries for each dice roll, one before and after, or try and find a way to combine them into a single line for efficiency's sake

**Comment by Apexal** [comment]
Since Firebase Firestore handles synchronizing seamlessly, we can do the following:
1. Create a log item in the log collection when a roll is started and hold on to its ID
2. When the roll completes, simply update the message of the log item with that ID

Everyone logged in will see the changes in realtime.

**Comment by Apexal** [comment]
A basic version of this has been implemented.
